src/core.ts

```diff
@@ -373,12 +374,23 @@ function TrackerCore(schemas: Schemas, sanitizeEvent: SanitizeEventFn): TrackerF
                 return getUnique(removedFields);
             },
 
+            getProtocol: (document) => {
+                const protocol = (document && document.location && document.location.protocol) || 'https';
+
+                // Only allowed protocols. Default to https
+                if (protocol !== 'http' && protocol !== 'https') {
+                    return 'https';
+                }
+
+                return protocol;
+            },
+
             /**
              * Use pixel endpoint to upload string to event-tracker
              * @param segment
              */
             sendSegment: (segment) => {
-                const protocol = (document && document.location && document.location.protocol) || 'https';
+                const protocol = internals.getProtocol(document);
```

Comment:
Returned string has no `:`. Needs to either add the `:` character here or have it returned.